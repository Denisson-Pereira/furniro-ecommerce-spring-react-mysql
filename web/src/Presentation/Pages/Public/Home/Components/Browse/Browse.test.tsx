import { render, screen } from "@testing-library/react";
import { I18nextProvider } from "react-i18next";
import { describe, it } from "vitest";
import i18n from "../../../../../Translate";
import { BrowserRouter } from "react-router-dom";
import { Browse } from ".";

describe("Browse Component", () => {
    it("", async () => {
        render(
            <BrowserRouter>
                <I18nextProvider i18n={i18n}>
                    <Browse />
                </I18nextProvider>
            </BrowserRouter>
        );

        expect(screen.getByText(/Browse the Range/)).toBeInTheDocument();
        const text2 = await screen.findByText(/Living/);
        expect(text2).toBeInTheDocument();
    })
})