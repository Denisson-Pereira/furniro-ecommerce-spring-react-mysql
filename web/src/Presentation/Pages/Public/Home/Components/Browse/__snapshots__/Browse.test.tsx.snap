// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Browse Component > should render correctly and match snapshot 1`] = `
<div>
  <div
    class="browse_container"
  >
    <div
      class="browse_title"
    >
      <p>
        Browse the Range
      </p>
      <span>
        Explore a wide range of products by category and unlock limitless opportunities for growth and success.
      </span>
    </div>
    <div
      class="browse_categories"
    >
      <div
        class="browse_container_img"
      >
        <img
          alt="Dining"
          src="https://firebasestorage.googleapis.com/v0/b/aracaju-carros.appspot.com/o/ecommerce%2Fcategories%2Fimage%20106%20(1).png?alt=media&token=f6c69d04-1464-45e9-800c-63f1bb054b9a"
        />
        <p>
          Dining
        </p>
      </div>
      <div
        class="browse_container_img"
      >
        <img
          alt="Living"
          src="https://firebasestorage.googleapis.com/v0/b/aracaju-carros.appspot.com/o/ecommerce%2Fcategories%2Fimage%20100.png?alt=media&token=f4c486fb-0b39-45a1-b89a-de2735b9d626"
        />
        <p>
          Living
        </p>
      </div>
      <div
        class="browse_container_img"
      >
        <img
          alt="Bedroom"
          src="https://firebasestorage.googleapis.com/v0/b/aracaju-carros.appspot.com/o/ecommerce%2Fcategories%2Fimage%20101.png?alt=media&token=5442a087-9265-401f-a0b8-cacfa82a886b"
        />
        <p>
          Bedroom
        </p>
      </div>
    </div>
  </div>
</div>
`;
